bpf/lib/conntrack.h

```diff
@@ -575,6 +575,107 @@ static __always_inline void ct4_cilium_dbg_tuple(struct __ctx_buff *ctx, __u8 ty
 	cilium_dbg(ctx, type, addr, rev_nat_index);
 }
 
+static __always_inline int
+ct_extract_ports4(struct __ctx_buff *ctx, int off, int dir,
+		  struct ipv4_ct_tuple *tuple, int *action,
+		  union tcp_flags *tcp_flags, bool *has_l4_header)
+{
+	int err;
+
+	switch (tuple->nexthdr) {
+	case IPPROTO_ICMP:
+		if (1) {
+			__be16 identifier = 0;
+			__u8 type;
+
+			if (ctx_load_bytes(ctx, off, &type, 1) < 0)
+				return DROP_CT_INVALID_HDR;
+			if ((type == ICMP_ECHO || type == ICMP_ECHOREPLY) &&
+			     ctx_load_bytes(ctx, off + offsetof(struct icmphdr, un.echo.id),
+					    &identifier, 2) < 0)
+				return DROP_CT_INVALID_HDR;
+
+			tuple->sport = 0;
+			tuple->dport = 0;
+
+			switch (type) {
+			case ICMP_DEST_UNREACH:
+			case ICMP_TIME_EXCEEDED:
+			case ICMP_PARAMETERPROB:
+				tuple->flags |= TUPLE_F_RELATED;
+				break;
+
+			case ICMP_ECHOREPLY:
+				tuple->sport = identifier;
+				break;
+			case ICMP_ECHO:
+				tuple->dport = identifier;
+				/* fall through */
+			default:
+				*action = ACTION_CREATE;
```

Comment:
In the current use of the function, `action` is not used. But I'm curious, why do we set `ACTION_CREATE` which I guess means create a new conntrack entry for types which are not `ICMP_ECHO`.